qt/scientific_interfaces/ISISReflectometry/GUI/Runs/RunsPresenter.cpp

```diff
@@ -143,14 +134,89 @@ void RunsPresenter::notifyTransfer() {
   transfer(m_view->getSelectedSearchRows(), TransferMatch::Any);
 }
 
-void RunsPresenter::notifyInstrumentChanged() { changeInstrument(); }
+void RunsPresenter::notifyInstrumentChanged() {
+  auto const instrumentName = m_view->getSearchInstrument();
+  if (m_mainPresenter)
+    m_mainPresenter->notifyInstrumentChanged(instrumentName);
+}
+
+void RunsPresenter::notifyInstrumentChanged(std::string const &instrumentName) {
+  m_mainPresenter->notifyInstrumentChanged(instrumentName);
+}
+
+void RunsPresenter::notifyReductionResumed() {
+  m_mainPresenter->notifyReductionResumed();
+}
+
+void RunsPresenter::notifyReductionPaused() {
+  m_mainPresenter->notifyReductionPaused();
+}
+
+void RunsPresenter::notifyAutoreductionResumed() {
+  m_mainPresenter->notifyAutoreductionResumed();
+}
+
+void RunsPresenter::notifyAutoreductionPaused() {
+  m_mainPresenter->notifyAutoreductionPaused();
+}
 
 void RunsPresenter::notifyStartMonitor() { startMonitor(); }
 
 void RunsPresenter::notifyStopMonitor() { stopMonitor(); }
 
 void RunsPresenter::notifyStartMonitorComplete() { startMonitorComplete(); }
 
+void RunsPresenter::reductionResumed() {
+  updateWidgetEnabledState();
+  tablePresenter()->reductionResumed();
+}
+
+void RunsPresenter::reductionPaused() {
+  updateWidgetEnabledState();
+  tablePresenter()->reductionPaused();
+}
+
+/** Searches ICAT for runs with given instrument and investigation id, transfers
+ * runs to table and processes them. Clears any existing table data first.
+ */
+void RunsPresenter::autoreductionResumed() {
+  if (requireNewAutoreduction()) {
+    // If starting a brand new autoreduction, delete all rows / groups in
+    // existing table first
+    // We'll prompt the user to check it's ok to delete existing rows
+
+    // tablePresenter()->setPromptUser(false);
+    // try {
+    //  tablePresenter()->notify(DataProcessorPresenter::DeleteAllFlag);
+    //} catch (const DataProcessorPresenter::DeleteAllRowsCancelledException &)
+    //{
+    //  return;
+    //}
+  }
+
+  if (m_autoreduction->setupNewAutoreduction(m_view->getSearchString()))
+    checkForNewRuns();
+
+  updateWidgetEnabledState();
+  tablePresenter()->autoreductionResumed();
+}
+
+void RunsPresenter::autoreductionPaused() {
+  m_view->stopTimer();
+  m_autoreduction->stop();
+  updateWidgetEnabledState();
+  tablePresenter()->autoreductionPaused();
+}
+
+void RunsPresenter::instrumentChanged(std::string const &instrumentName) {
+  UNUSED_ARG(instrumentName)
```

Comment:
does the `UNUSED_ARG` flag need to be there since we're passing instrumentName into `setSearchIntrument` and `instrumentChanged`?

Reply:
No - I'll take it out :+1: